### DISCUSSION TITLE
Backup Calling Guide

### DISCUSSION THREAD

**Discussion by fillwithjoy1** [kyujin-cho/pixel-volte-patch#272] [General]
This is a guide for activating backup calling on your device. Android 14
TL:DR your calling SIM should have "Switch mobile data automatically" turned on. You do not need to turn on Data or Roaming. If the Calling SIM then reads "(carrier name) Backup calling" even if it reports no signal then Backup Calling is on. At this point just check that your data SIM allows wifi calling (if VoLTE works, you should be fine) and then pray that it connects. 

For me it is not reliable and I'm occasionally able to get it to work but the success **varies heavily based on your country. Just because it's unreliable for me doesn't mean that it won't work for you**

Long guide:
Clearer guide on what to do (eSIM, pSIM doesn't matter)
When I refer to the `Data SIM` we are talking about the local sim that has internet available
When I refer to the `Calling SIM` we are talking about the roaming sim that has Wi-Fi Calling available (VoWIFI). If your carrier didn't have Wi-Fi calling working (test with airplane mode) before you travelled, this won't work.

1. On your `Calling SIM` enable these toggles
- `Enable VoLTE`
- `Enable Backup Calling`
- `Enable VoWiFi`
- `Enable VoWiFi while roaming`

cosmetic wise, you can enable these
- `Show VoWiFi preference in Settings`
- `Show VoWiFi Roaming preference in Settings`
- `Show VoWiFi Icon`

2. Once you have done these go to your `Calling SIM` settings and enable `Switch mobile data automatically` (screenshot provided)
You **don't need to enable Roaming or Mobile data**
![Screenshot_20240224-192317](https://github.com/kyujin-cho/pixel-volte-patch/assets/29514446/92ba0541-a267-438c-a7fd-42124a2aee53)

3. Restart your phone
4. You should observe your `Calling SIM` showing "Backup calling". It doesn't have to be connected yet (screenshots below, my `Data SIM` is Jio and you'll notice calling anyone with `Calling SIM` reports no signal).
![Screenshot_20240224-192339](https://github.com/kyujin-cho/pixel-volte-patch/assets/29514446/f075e147-a7f2-43ea-9392-eacade30fb07)
![Screenshot_20240224-192921](https://github.com/kyujin-cho/pixel-volte-patch/assets/29514446/dcbefcc0-385b-4c47-b854-088499e4d929)

6. Eventually your Calling SIM should connect successfully.

# Troubleshooting
## My `Calling SIM` is stuck on "Backup calling" but doesn't connect in the end.
There could be a variety of reasons why it doesn't work. Personally I've found that opening Pixel IMS and going to `Expert Mode` on my `Calling SIM` and turning on `FORCE_HOME_NETWORK_BOOL` has started to get it working. 

Edit: It can also be due to your network blocking Wi-Fi calling (UDP Port 4500, IPsec Pass-Through) [^2]

## My `Calling SIM` switches to "No Service" / "Emergency calls only" after a while (but it did connect successfully before)
Try this:
1. Open Pixel IMS
2. Go to your `Calling SIM`
3. Go to `Expert Mode`
4. Search for `FORCE_HOME_NETWORK_BOOL` and set it to true
5. Restart your device

## My `Calling SIM` is saying it's connected to Roaming and yet reports that it's connected as Backup calling.
**You have Roaming On and it's overriding Backup calling**
1. Go to your `Calling SIM` SIM card page and scroll down to Network section
7. Under there click on `Select Network`, or toggle off `Automatically Select Network`.
8. Notice that your `Calling SIM` is connected to a local network operator. 
9. Click on a network operator that has been marked as `(forbidden)`. This will force your phone to use Backup Calling as it sees that Roaming is now not available.
10. Observe your network operator change to Backup calling

## My `Data SIM` is marked as Roaming, will this cause issues?
**Yes because backup calling only kicks in if your `Data SIM` is in the "home network" (aka no roaming)** [^1]
For Backup Calling to kick in, your `Calling SIM` has to be in Roaming or Out of Service **and** your `Data SIM` has to be in the "home network". If your `Data SIM` is marked as roaming, then you'll need to bypass this by:
- Open Pixel IMS and select your data sim
- Open Expert Mode
- Search up `FORCE_HOME_NETWORK_BOOL` and set it to True

I am available for anyone who needs to get this working, just open a new thread and I'll do my best to help you

# Outro
It's finicky at times. YMMV. I've noticed that Backup Calling works more reliably with different SIMs and even with different countries, so go figure

Edit: Your carrier plays a huge role. For me my carrier requires me to stay connected to a tower to even enable wi-fi calling, making it nearly impossible to use backup calling effectively. I've got another carrier that doesn't have the same restriction, but keep that in mind

# Notes
 [^1]: [Conditions for Backup calling to kick in](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/telephony/java/android/telephony/ims/ImsMmTelManager.java;l=1014-1047)
[^2]: [Wi-Fi Calling ports required](https://www.telstra.com.au/support/mobiles-devices/telstra-wifi-calling#:~:text=Will%20my%20modem%20work%20with%20Wi-Fi%20Calling?)

**Comment by deltaxsingh** [comment]
I'm trying this on my Samsung S22 Android 14. Nada, expert mode and all. 

Wi-Fi calling works fine but as soon as it switches over to data, it's done. Not sure wh Google can't get basic stuff to work compared to Apple.

Calling sim: Roaming
Data sim: Local

**Reply by fillwithjoy1** [reply]
To be honest, it's finicky at times. I've observed backup calling working more reliably with different SIM cards and different countries. YMMV

**Comment by choozuradventure** [comment]
@fillwithjoy1 Forgive my ignorance, but is this patch necessary if you already have the "Switch mobile data automatically" toggle available under your calling SIM?  Or is this patch necessary to get that toggle to show up in the first place?

**Reply by fillwithjoy1** [reply]
Great question. Let me explain what happens when you connect your phone to your carrier for the first time

(not accurate, this is to the best of my knowledge)

A. Your phone identifies cell towers nearby
B. Using your SIM card, your phone determines which towers it can connect to (to use your carrier services)
C. Once successfully connected, your device then talks to the carrier to download the latest configuration files from your carrier

This configuration file is important. It tells your phone where voicemail exists, which data services are allowed and more importantly features that you can access such as Wi-Fi Calling, 5G Calling, etc.

Issue is, if you wanted to manually force 5G Calling you couldn't get past this. **This is what Pixel IMS does!** It gives you the ability to edit this configuration file. As you probably suspected, Backup calling can be turned off by your carrier. That's why I've included that.

**Comment by Paqueta7** [comment]
My calling card  (Malaysia) can make calls on flight mode with wifi connected with any phones (Samsung) in my country (Malaysia).
It displays my network provider (U Mobile) in flight mode with wifi connected.

**First Scenario**
I have a Pixel 7 from Japan, my calling sim (Malaysia) cannot make call on flight mode with wifi connected,
In Pixel 7 when I put it  on flight mode with wifi connected, 
It shows like pop up saying **"Turn off airplane mode or connect to a wireless network to make call"**
It does NOT displays my network provider too.
Refer the image below
![1000160438](https://github.com/user-attachments/assets/6f2cbf0d-dee3-4466-a9fd-4aacf780b14b)

**Second Scenario**
When I turn off flight mode keep it conected to my wifi, it can call as usual and there is a small note *Wi-Fi call* on my dialer.
Refer the image below
![1000160445](https://github.com/user-attachments/assets/730278f0-cff2-4cca-9f01-d51f4797dfe9)

Currently travelling in Brazil, my calling sim is still from Malaysia and my data sim from Brazil.
It not connecting most of the time in backup calling, need help to resolve making call uisng flight mode issue and back calling. 
Please advice/urgent help needed, I still want to use Pixel 7 because of my passion for Pixel and it was a gift.

**Reply by scrag** [reply]
I’m pretty sure you need to make sure your Calling Sim (Malaysia) has no connection. Attempt to connect to a forbidden network for the Calling Sim and ensure “switch mobile data automatically” is enabled for it. It should then work.

**Reply by fillwithjoy1** [reply]
Interesting. What I usually do is turn on airplane mode and then pull down the QS panel, and wait. Generally after some time, I see myself connecting to my provider through Wi-Fi calling. It's worked so well for me that I've used tasker to turn airplane mode at home to save power since I can call and text on Wi-Fi.

TL:DR Try again, and wait. After some time it should connect to your provider through Wi-Fi calling. If you want to be notified, try texting that number and then as soon as your SIM connects, you'll be notified as an SMS will then come through

**Comment by Paqueta7** [comment]
Thanks @scrag for your feedback, but it will drain the battery and get the phone heated if it continuesly keep finding for signal....it defeats the purpose, any other way?

**Comment by scrag** [comment]
It will get a signal eventually through a.partner network. I've got it working that way without issue when overseas.

This is how backup dialling works.


________________________________

**Comment by Paqueta7** [comment]
You mean no connection like flight mode? Once before going to the new android version. When I turn on flight mode and WiFi on, it showed my carrier in Malaysia. Then after that it didn't and I can't do any flight mode calls with WiFi turn on.

**Comment by scrag** [comment]
On your Calling Sim you need to select a network that is forbidden so that the Calling Sim has no connection.


________________________________

**Comment by Paqueta7** [comment]
Oh ok, I get it....anyway to use back calling from travelling sim data?

**Comment by scrag** [comment]
That's what backup calling is.

You are making calls from your Calling SIM through the data connection of your Data SIM.

When you make a call it should say something like "calling via 'Calling SIM' shared-data call".



________________________________

**Comment by fillwithjoy1** [comment]
@Paqueta7 and @scrag Can we please discuss under this thread. It's making this discussion look very messy.

@Paqueta7 Have you read my guide completely? It tells you what you need to configure to get backup calling working. See if you can get Wi-Fi calling working first.

**Comment by Paqueta7** [comment]
Yes @fillwithjoy1 , I have read your guide. but  my phone is not working in vowifi, if I turn on airplane mode and and wifi. If works well if i use turn off airplane mode and keep my wifi on, I'm able to call my home country. And also your guide need to be updated a little where i think most people have moved to android 15 and the options are diffrent. Please advice,

**Reply by fillwithjoy1** [reply]
> Yes @fillwithjoy1 , I have read your guide. but my phone is not working in vowifi, if I turn on airplane mode and and wifi. If works well if i use turn off airplane mode and keep my wifi on, I'm able to call my home country. And also your guide need to be updated a little where i think most people have moved to android 15 and the options are diffrent. Please advice,

Then it's likely not connecting at all, and your carrier is to blame. If you can't get it to work with Airplane Mode then you can't get Backup Calling working either.

**Reply by Paqueta7** [reply]
Thanks opinion, just to let you know. The carrier which is Umobile, works on other phones such Xiomi, Samsung and many more on on flight mode with wifi, it is able to make calls....why pixel can't? is there any setting i missed?

**Reply by fillwithjoy1** [reply]
> Thanks opinion, just to let you know. The carrier which is Umobile, works on other phones such Xiomi, Samsung and many more on on flight mode with wifi, it is able to make calls....why pixel can't? is there any setting i missed?

Talk to your carrier about this. They likely haven't whitelisted your phone, unless you modify your IMS agent to look like another Samsung device

**Comment by chaosgeo2k** [comment]
does this backup calling still works on Android 15?  if yes can the guide be updated 

Q#2: does the Data SIM really need wifi and\or voLTE enabled?   it seems that it would only use the data part to make a call.

**Reply by fillwithjoy1** [reply]
1. Just imagine that the guide has been updated to work for Android 15 (there's nothing new to write)
2. Yes. You don't need mobile data to be on, but enable VoLTE and Wi-Fi Calling.

**Comment by iletired** [comment]
This guide is a literal savior and I share it every time someone asks about using a foreign data sim to act as WiFi calling on a pixel.
Niche problem at work though, I'm wondering if you can help solve in the app @kyujin-cho  - I've played around with the values and did some searching and at a loss.

USA based here. Two SIMs. SIM 1 has no service at one job site. SIM 2 does, but is very slow, but it's enough for me to get backup calling and backup texting. Is there a bool where I remain connected to WiFi, but because wifi ports are blocked, it falls back to Backup Calling as SIM2 has data?

I work in the IT department, yet they are incompetent to fix the issue despite me pushing the issue the last two months since it started. I'm on a 9 Pro.

**Reply by fillwithjoy1** [reply]
> This guide is a literal savior and I share it every time someone asks about using a foreign data sim to act as WiFi calling on a pixel. Niche problem at work though, I'm wondering if you can help solve in the app @kyujin-cho - I've played around with the values and did some searching and at a loss.
> 
> USA based here. Two SIMs. SIM 1 has no service at one job site. SIM 2 does, but is very slow, but it's enough for me to get backup calling and backup texting. Is there a bool where I remain connected to WiFi, but because wifi ports are blocked, it falls back to Backup Calling as SIM2 has data?
> 
> I work in the IT department, yet they are incompetent to fix the issue despite me pushing the issue the last two months since it started. I'm on a 9 Pro.

Hey, thanks for letting me know that the guide is extremely helpful. It's good to know that this is now being used as a resource for people to use. As for the Wi-Fi issue, I don't think it's possible at all. Here I've pulled a screenshot directly from Android's codebase (check note 1 for link)
![image](https://github.com/user-attachments/assets/6e3d702b-a6b7-4e13-82eb-b81ee6b1c5c7)

Maybe if you disable Wi-Fi, but then you'd be using up all your data so that's not ideal. You can also try using a VPN to circumvent the problem completely.

**Comment by yorner1** [comment]
European Samsung Galaxy S24 Ultra user here, tried everything but unfortunately nothing works to get backup calling working, it simply only works on WiFi

When I desconnect from WiFi it reverts back to "Emergency calls only"

**Reply by fillwithjoy1** [reply]
It's likely your carrier settings are forcing the device to **only use wi-fi calling when you are in range of your carrier's towers**

I do need to investigate my carrier's settings and figure out what's wrong but that's what I'm thinking

**Reply by yorner1** [reply]
The calling E-SIM is a US AT&T line, while the data SIM is European from Vodafone, if that helps

**Comment by jabashque** [comment]
So, I know that the actual AOSP codebase specifically mentions that the data sim needs to be in the home network for Backup Calling to work, hence the reason why you have a section in your Troubleshooting section dedicated to talking about the scenario where your data sim is roaming. However, I have tested the following scenarios:
* T-Mobile US as my voice sim (forced to connect to a network that it can't connect to, so it's in a disconnected state) and Plus as my data sim in the US, where my Plus sim is roaming on Verizon and AT&T
* T-Mobile US as my voice sim (forced to connect to a network that it can't connect to, so it's in a disconnected state) and Drei (3 Austria) as my data sim in the US, where my Drei sim is roaming on Verizon and T-Mobile

And in both situations, my Pixel 9 Pro XL (April 2025 patch) still actually establishes Backup Calling for my T-Mobile US sim, despite my data sim being on roaming. I don't need to force override `FORCE_HOME_NETWORK_BOOL` to true at all.

It makes me wonder if Google forgot to update the code comments to reflect this new behavior...

**Reply by fillwithjoy1** [reply]
🤦🤦 If it's not obvious, your carrier's config file impacts whether you are able to connect or not. This is nothing new that you discovered, only that your carrier plays nice.

**Reply by jabashque** [reply]
Yeah, no, that's not what I'm referring to. To quote what you have in your Troubleshooting section:

> My Data SIM is marked as Roaming, will this cause issues?
> Yes because backup calling only kicks in if your Data SIM is in the "home network" (aka no roaming)

I already stated that in both situations, my data sim is in a roaming state and not connected to its home network (i.e. I'm in the US and the home network of the two Data SIMs I tried are Plus (Poland) and Drei (Austria)). Even though the carrier config file for T-Mobile US (Calling SIM) has cross sim calling enabled, my phone is still not supposed to actually do cross sim calling because my Data SIM is roaming and not connected to the home network, hence why you have an entire section in your Troubleshooting section talking about needing to enable `FORCE_HOME_NETWORK_BOOL` for the Data SIM if it's roaming.

Now, what I am trying to say is that it seems that for me, my phone happily did cross sim calling even thought my Data SIM was roaming and not connected to its home network AND without `FORCE_HOME_NETWORK_BOOL` being enabled at all on my Data SIM, suggesting that the code comments that Google left in the AOSP codebase about this are no longer accurate.

**Reply by raywdude** [reply]
What I also found out is Pixel 9 works perfectly with backup calling in just about every combination possible in my testing as well.

However I also have a Pixel 8 Pro (I use exclusively for testing and development ) that I just wiped and clean installed the latest Android 15 official factory images May 2025 security patch and...backup calling just wouldn't work with any combination I do.

Everything is turned on - wifi calling, automatic data switching, on every sim. Same setup as on the Pixel 9.

No dice.

ATT as the calling sim and T-Mobile as the data sim....no go. Other way around? No go.
Verizon? No go.


Pixel 9 doesn't care. Every combination thereof worked perfectly.

**Comment by cipherdelion** [comment]
Hi @fillwithjoy1 , I am using Evolution X based on Android 16 on my Redmi Note 10 Pro in India. I want to enable backup calling feature for both of my SIMs to fix the calling issue when one SIM lost coverage. Is it possible to achieve this using this Pixel IMS workaround? I am new to these things.
Thanks.

**Comment by No-WaY-SA** [comment]
Hello any way to activate auto switch data sim option? I have OnePlus open Apex edition android 15 and this option is not available in EU version ,but in IN is there, I already activated backup calling but still not working and I believe it because auto switch data sim not active.
